typed/merge_test.go

```diff
@@ -423,6 +467,18 @@ var mergeCases = []mergeTestCase{{
 		`{"atomicList":["a","a","a"]}`,
 		`{"atomicList":["a","a"]}`,
 		`{"atomicList":["a","a"]}`,
+	}, {
+		`{"list":[{"key":"a","id":1,"bv":true},{"key":"b","id":2},{"key":"a","id":1,"bv":false,"nv":2}]}`,
+		`{"list":[{"key":"a","id":1,"nv":3}, {"key":"c","id":3},{"key":"b","id":2}]}`,
+		`{"list":[{"key":"a","id":1,"bv":true,"nv":3},{"key":"c","id":3},{"key":"b","id":2}]}`,
```

Comment:
So this is merging the new item with the first occurence, I'm currently thinking we should replace altogether.

Reply:
This case surprised me. From our prior conversations I would not have expected the merge with one of the duplicates. It doesn't seem like as the author of such a resource I'd be very conscious of which of the duplicates I am merging with.


Is it possible to have two instances of the same value but they are owned by different field managers? In that case I almost would expect "MY" version of the duplicate to be merged; but don't think that'd be worth the effort.

Reply:
> Is it possible to have two instances of the same value but they are owned by different field managers?

No.

> This case surprised me. From our prior conversations I would not have expected the merge with one of the duplicates. It doesn't seem like as the author of such a resource I'd be very conscious of which of the duplicates I am merging with.

yeah, I think we just need to replace rather than merge, which means I'll need to know if they are duplicated and replace altogether, I'll update the algorithm.